src/dvclive/studio.py

```diff
@@ -81,3 +82,11 @@ def post_to_studio(live, event_type, logger) -> bool:
     )
 
     return response.status_code == 200
+
+
+def get_studio_repo_url(git_repo):
+    # https://github.com/iterative/studio/blob/main/backend/repos/models/repo.py#L67
+    remote_name = git_repo.active_branch.tracking_branch().remote_name
+    remote_url = git_repo.remotes[remote_name].url
+    parsed = urlparse(remote_url.replace(".git", ""))
+    return f"{parsed.netloc}:{parsed.path.lstrip('/')}"
```

Comment:
Please have look at [this](https://github.com/iterative/studio/blob/develop/backend/repos/urltools.py) file. We have [these](https://github.com/iterative/studio/blob/develop/backend/repos/urltools.py#L13-L17) types of remotes + `custom-gitlab` (can be handled with `STUDIO_REPO_URL` for now). We strip down `.com` from them.

Overall suggestion, let's validate remotes against whitelisted URLs (main GH/GL/BB SaaS deployments). Otherwise, we should fail and ask the user to explicitly provide `STUDIO_REPO_URL`.

@nimdraugsael let us know if it makes sense.

Reply:
> Please take a look at [this](https://github.com/iterative/studio/blob/develop/backend/repos/urltools.py) file. We have [these](https://github.com/iterative/studio/blob/develop/backend/repos/urltools.py#L13-L17) types of remotes + custom-gitlab (can be handled with STUDIO_REPO_URL for now). We strip down .com from them.

Skimmed through the file and missed some parts.
Added tests for the 3 providers and ssh.

> let's validate remotes against allowlisted URLs (main GH/GL/BB SaaS deployments). Otherwise, we should fail and ask the user to explicitly provide STUDIO_REPO_URL. If we don't do this, then gitlab.freedesktop.org's hosted repo will become gitlab.freedesktop.org:iterative/dvclive-example, which is invalid from Studio BE PoV.

Not sure I follow why we should duplicate that sort of validation on the DVCLive side. There are other cases where we rely on Studio backend to handle the invalid request and return the appropriate response (invalid token, non existing repo, etc).
We already handle those responses from studio and log them in debug mode.

Reply:
> Not sure I follow why we should duplicate that sort of validation on the DVCLive side

I believe currently we'll receive ["401 Permission denied"](https://github.com/iterative/studio/blob/develop/backend/live/views.py#L57) from the Studio backend on an invalid URL, which does not explain what is the problem and how to mitigate it. BE does not validate such URLs anyhow, except [checking that it is a string](https://github.com/iterative/studio/blob/develop/backend/live/views.py#L32).

Validating these things on the DVCLive side makes us fail fast and say which env variable the user should modify to make it work. Making Studio BE know that DVCLive uses the `STUDIO_REPO_URL` env variable and where are the docs for it to build a correct error message does not seem to be the best solution IMO. We may have another consumer of Studio's live ingestion endpoint and for them, the message `Please modify STUDIO_REPO_URL as ${docsLink}` may not make a lot of sense as `STUDIO_REPO_URL` exists only in the context of DVCLive.

Reply:
Added some validation and warnings. When docs are ready I will update to include the link in the message

Reply:
Still feel like Studio validation could give more info beyond Permission denied. If I understand correctly, passing a non existing `repo_url` will return the same error as passing a valid `repo_url` with an invalid token.

Reply:
I agree with you, it should be done on both sides. I've created https://github.com/iterative/studio/issues/4736 to iterate on it.

Reply:
> When docs are ready I will update to include the link in the message

Do we need an issue for it to not forget/add it to our [shared checklist](https://github.com/iterative/studio/issues/4666)?